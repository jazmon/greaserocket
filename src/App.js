// @flow
import React from 'react';
import { AppRegistry, Platform } from 'react-native';
import codePush from 'react-native-code-push';
import Raven from 'raven-js';
import ravenRN from 'raven-js/plugins/react-native';
import { Provider } from 'react-redux';
import { TabNavigator, StackNavigator } from 'react-navigation';
import { ThemeProvider } from 'styled-components';
import type { Store } from 'redux';
import type { ReduxState } from 'redux/modules';
import type { Action } from 'types';

import configureStore from 'store/configureStore';
import config from 'constants/config';
import Feed from 'screens/Feed';
import Profile from 'screens/Profile';
import MapScreen from 'screens/Map';
import EventDetail from 'screens/EventDetail';
import theme from 'constants/theme';

ravenRN(Raven);
const App = TabNavigator(
  {
    Feed: {
      screen: Feed,
      path: 'feed',
    },
    Profile: {
      screen: Profile,
      path: 'profile/:name',
    },
    Map: {
      screen: StackNavigator({
        Map: {
          screen: MapScreen,
        },
        EventDetail: {
          screen: EventDetail,
          path: 'eventdetail/:eventId',
        },
      }),
      path: 'map',
    },
  },
  {
    animationEnabled: true,
    swipeEnabled: true,
    lazyLoad: true,
    screenProps: {
      foo: 'bar',
    },
    initialTab: 'Feed',
    tabBarOptions: {
      activeTintColor: '#e91e63',
    },
    containerOptions: {
      // on Android, the URI prefix typically contains a host in addition to scheme
      URIPrefix: Platform.OS === 'android'
        ? `${config.NAVIGATION.URI_PREFIX}://${config.NAVIGATION.URI_PREFIX}/`
        : `${config.NAVIGATION.URI_PREFIX}://`,
    },
  }
);

const store: Store<ReduxState, Action<*>> = configureStore();
const Root = () => (
  <Provider store={store}>
    <ThemeProvider theme={theme}>
      <App />
    </ThemeProvider>
  </Provider>
);

const codePushOptions = {
  checkFrequency: codePush.CheckFrequency.ON_APP_RESUME,
};

const WrappedRoot = codePush(codePushOptions)(Root);
AppRegistry.registerComponent('Greaserocket', () => WrappedRoot);

export const AppComponent = Root;
export default WrappedRoot;
