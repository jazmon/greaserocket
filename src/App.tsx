import React from 'react';
import { AppRegistry, Platform } from 'react-native';
// import codePush from 'react-native-code-push';
import { Provider } from 'react-redux';
import { TabNavigator, StackNavigator } from 'react-navigation';
import { ThemeProvider } from 'styled-components';
import { Store } from 'redux';
import { ReduxState } from 'redux/modules';
import { Action } from 'types';

import configureStore from 'redux/store/configureStore';
import config from 'constants/config';
import Feed from 'screens/Feed';
import Profile from 'screens/Profile';
import MapScreen from 'screens/Map';
import Messages from 'screens/Messages';
import EventDetail from 'screens/EventDetail';
import theme from 'constants/theme';

import { graphql, ApolloProvider } from 'react-apollo';

import ApolloClient, { createNetworkInterface } from 'apollo-client';

const App = TabNavigator(
  {
    Feed: {
      screen: Feed,
      path: 'feed',
    },
    Messages: {
      screen: Messages,
      path: 'messages',
    },
    Profile: {
      screen: Profile,
      path: 'profile/:name',
    },
    Map: {
      screen: StackNavigator({
        Map: {
          screen: MapScreen,
        },
        EventDetail: {
          screen: EventDetail,
          path: 'eventdetail/:eventId',
        },
      }),
      path: 'map',
    },
  },
  {
    animationEnabled: true,
    swipeEnabled: true,
    // lazyLoad: true,
    screenProps: {
      foo: 'bar',
    },
    initialTab: 'Feed',
    tabBarOptions: {
      activeTintColor: '#e91e63',
    },
    // on Android, the URI prefix typically contains a host in addition to scheme
    uriPrefix:
      Platform.OS === 'android'
        ? `${config.NAVIGATION.URI_PREFIX}://${config.NAVIGATION.URI_PREFIX}/`
        : `${config.NAVIGATION.URI_PREFIX}://`,
  },
);

const store: Store<ReduxState> = configureStore();
const Root = () =>
  <Provider store={store}>
    <ThemeProvider theme={theme}>
      <App />
    </ThemeProvider>
  </Provider>;

// const codePushOptions = {
//   checkFrequency: codePush.CheckFrequency.ON_APP_RESUME,
// };

// const WrappedRoot = codePush(codePushOptions)(Root);
AppRegistry.registerComponent('Greaserocket', () => Root);

export const AppComponent = Root;
// export default Root;
