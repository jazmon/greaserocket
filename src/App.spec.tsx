import 'react-native';
import * as React from 'react';
// Note: test renderer must be required after react-native.
import * as renderer from 'react-test-renderer';
import { AppComponent } from './App';

it('renders correctly', () => {
  const tree = renderer.create(<AppComponent />);
  expect(tree).toMatchSnapshot();
});
