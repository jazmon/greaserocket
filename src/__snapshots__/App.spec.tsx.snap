// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  loaded={
    Array [
      0,
    ]
  }
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    DEPRECATED_sendUpdatedChildFrames={false}
    alwaysBounceHorizontal={false}
    alwaysBounceVertical={false}
    automaticallyAdjustContentInsets={false}
    bounces={false}
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
    contentOffset={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    directionalLockEnabled={true}
    horizontal={true}
    keyboardDismissMode="on-drag"
    keyboardShouldPersistTaps="always"
    onContentSizeChange={null}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={undefined}
    onResponderTerminationRequest={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onScrollShouldSetResponder={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    pagingEnabled={true}
    scrollEnabled={true}
    scrollEventThrottle={16}
    scrollsToTop={false}
    sendMomentumEvents={true}
    showsHorizontalScrollIndicator={false}
    style={
      Array [
        Object {
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "overflow": "scroll",
        },
        Object {
          "flexGrow": 1,
        },
      ]
    }
  >
    <RCTScrollContentView
      collapsable={false}
      removeClippedSubviews={undefined}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "flexGrow": 1,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "overflow": "hidden",
          }
        }
        testID={undefined}
      >
        <View
          style={
            Object {
              "flex": 1,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#FFFFFF",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "flexBasis": 0,
                    "flexDirection": "column",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "marginTop": 24,
                  },
                  undefined,
                ]
              }
            >
              <RCTScrollView
                DEPRECATED_sendUpdatedChildFrames={false}
                alwaysBounceHorizontal={undefined}
                alwaysBounceVertical={true}
                dataSource={
                  ListViewDataSource {
                    "items": 0,
                  }
                }
                enableEmptySections={true}
                onContentSizeChange={null}
                onMomentumScrollBegin={[Function]}
                onMomentumScrollEnd={[Function]}
                onResponderGrant={[Function]}
                onResponderReject={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={undefined}
                onResponderTerminationRequest={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                onScrollShouldSetResponder={[Function]}
                onStartShouldSetResponder={[Function]}
                onStartShouldSetResponderCapture={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                pageSize={6}
                refreshControl={
                  <RefreshControlMock
                    onRefresh={[Function]}
                    refreshing={false}
                  />
                }
                renderRow={[Function]}
                renderScrollComponent={[Function]}
                renderSectionHeader={[Function]}
                renderSeparator={[Function]}
                scrollEventThrottle={undefined}
                sendMomentumEvents={false}
                style={
                  Array [
                    Object {
                      "flexDirection": "column",
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "overflow": "scroll",
                    },
                    undefined,
                  ]
                }
              >
                <RCTRefreshControl />
                <RCTScrollContentView
                  collapsable={false}
                  removeClippedSubviews={undefined}
                  style={
                    Array [
                      undefined,
                      undefined,
                    ]
                  }
                />
              </RCTScrollView>
            </View>
          </View>
        </View>
      </View>
      <View
        style={null}
        testID={undefined}
      />
      <View
        style={null}
        testID={undefined}
      />
      <View
        style={null}
        testID={undefined}
      />
    </RCTScrollContentView>
  </RCTScrollView>
  <View
    collapsable={false}
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#f4f4f4",
          "borderTopColor": "rgba(0, 0, 0, .2)",
          "borderTopWidth": NaN,
          "flexDirection": "row",
          "height": 49,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "stretch",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "flex": 1,
            "justifyContent": "flex-end",
          }
        }
        testID={undefined}
      >
        <View
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#e91e63",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#929292",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          collapsable={undefined}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgba(233, 30, 99, 1)",
              "fontSize": 10,
              "marginBottom": 1.5,
              "textAlign": "center",
            }
          }
        >
          Feed
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "stretch",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "flex": 1,
            "justifyContent": "flex-end",
          }
        }
        testID={undefined}
      >
        <View
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          collapsable={undefined}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgba(146, 146, 146, 1)",
              "fontSize": 10,
              "marginBottom": 1.5,
              "textAlign": "center",
            }
          }
        >
          Messages
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "stretch",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "flex": 1,
            "justifyContent": "flex-end",
          }
        }
        testID={undefined}
      >
        <View
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#e91e63",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#929292",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          collapsable={undefined}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgba(146, 146, 146, 1)",
              "fontSize": 10,
              "marginBottom": 1.5,
              "textAlign": "center",
            }
          }
        >
          Profile
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "stretch",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "flex": 1,
            "justifyContent": "flex-end",
          }
        }
        testID={undefined}
      >
        <View
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#e91e63",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#929292",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          collapsable={undefined}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgba(146, 146, 146, 1)",
              "fontSize": 10,
              "marginBottom": 1.5,
              "textAlign": "center",
            }
          }
        >
          Map
        </Text>
      </View>
    </View>
  </View>
</View>
`;
